FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -143,14 +144,16 @@ + (FIRMessagingExtensionHelper *)extensionHelper {
 }
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"
-- (instancetype)initWithAnalytics:(nullable id<FIRAnalyticsInterop>)analytics
-                 withUserDefaults:(GULUserDefaults *)defaults {
+- (instancetype)initForApp:(FIRApp *)app
```

Comment:
Instead of injecting the app here - it's likely better / less setup for tests to pass in the `heartbeatLogger` directly. Although seems like this API isn't used in tests since no tests changed that depend on it?

Reply:
Also, `initWith` is preferred over `initFor`

Reply:
> @ryanwilson: Instead of injecting the app here - it's likely better / less setup for tests to pass in the heartbeatLogger directly. Although seems like this API isn't used in tests since no tests changed that depend on it?

Correct. But I can change the initializer to `initWithHeartbeatLogger:analytics:userDefaults:` anyhow.

Reply:
Reworked in a3dc31a